redash/query_runner/athena.py

```diff
@@ -130,28 +135,41 @@ def __get_schema_from_glue(self):
         database_paginator = client.get_paginator('get_databases')
         table_paginator = client.get_paginator('get_tables')
 
-        for databases in database_paginator.paginate():
-            for database in databases['DatabaseList']:
-                iterator = table_paginator.paginate(DatabaseName=database['Name'])
-                for table in iterator.search('TableList[]'):
-                    table_name = '%s.%s' % (database['Name'], table['Name'])
-                    if table_name not in schema:
-                        column = [columns['Name'] for columns in table['StorageDescriptor']['Columns']]
-                        schema[table_name] = {'name': table_name, 'columns': column}
-                        for partition in table.get('PartitionKeys', []):
-                            schema[table_name]['columns'].append(partition['Name'])
+        def __db_query():
```

Comment:
Looks better. I still think that there's a lot of detail in `get_schema ` and `__get_schema_from_glue`, and moving these functions outside will help make them simpler.